File: nvflare/edge/web/models/job_response.py
from typing import Optional

from nvflare.edge.web.models.base_model import BaseModel


class JobResponse(BaseModel):

    def __init__(
        self,
        status: str,
        job_id: str = None,
        job_name: str = None,
        method: str = None,
        job_data: Optional[dict] = None,
        retry_wait: int = None,
        **kwargs,
    ):
        super().__init__()
        self.status = status
        self.job_id = job_id
        self.job_name = job_name
        self.method = method
        self.job_data = job_data
        self.retry_wait = retry_wait

        if kwargs:
            self.update(kwargs)

File: nvflare/edge/device/android/app/src/main/java/com/nvidia/nvflare/sdk/models/JobResponse.kt
package com.nvidia.nvflare.sdk.models

import com.google.gson.annotations.SerializedName
import com.nvidia.nvflare.sdk.training.Job
import com.google.gson.JsonObject
import com.nvidia.nvflare.sdk.core.NVFlareError

data class JobResponse(
    @SerializedName("status")
    val status: String,
    
    @SerializedName("job_id")
    val jobId: String?,
    
    @SerializedName("job_name")
    val jobName: String?,
    
    @SerializedName("job_data")
    val jobData: JsonObject?,
    
    @SerializedName("method")
    val method: String?,
    
    @SerializedName("retry_wait")
    val retryWait: Int?,
    
    @SerializedName("message")
    val message: String?,
    
    @SerializedName("details")
    val details: Map<String, String>?
) {
    fun toJob(): Job {
        if (jobId == null) {
            throw NVFlareError.InvalidRequest("Can't convert JobResponse to Job")
        }
        return Job(id = jobId, status = "running")
    }
}
